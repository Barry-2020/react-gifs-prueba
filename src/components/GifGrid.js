import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
// import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    // const [images, seImages] = useState([])
    const { data:images, loading } = useFetchGifs( category );



    // encodeURI reemplaza espacios formato url
    return (
        <>
        <h3 className='animate__animated animate__fadeInLeft'>{category}</h3>
        { loading && <p className='animate__animated animate__flash'>Loading...</p>}
        <div className='card-grid'>
                {
                    images.map(( img ) =>(
                        <GifGridItem 
                            key={ img.id } 
                            { ...img }
                        />
                    ))
                }
   
        </div>
        </>
        
    )
}
