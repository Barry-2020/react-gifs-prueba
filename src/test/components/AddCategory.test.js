import { shallow } from "enzyme"
import '@testing-library/jest-dom';
import { AddCategory } from "../../components/AddCategory"

describe('Pruebas adCategory', () => { 

    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={ setCategories }/>);
    beforeEach( () => {
        // lIMPIAMoS
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategories }/>);
    });

    test('Debe de mostrar correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'hola mundo';
        input.simulate('change', { target: { value }});
        expect( wrapper.find('p').text().trim() ).toBe( value );
    })

    test('no debe de postear info con submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategories ).not.toHaveBeenCalled();
    })

    test('Debe de llamar el setCategories y limpiar la caja de texto', () => {
        const value = 'Hola Mundo';

        wrapper.find('input').simulate('change', { target: { value }});
        
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledTimes(1);
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );
        // expect( setCategories ).toHaveBeenCalled();

        expect( wrapper.find('input').prop('value') ).toBe('');
    })
})