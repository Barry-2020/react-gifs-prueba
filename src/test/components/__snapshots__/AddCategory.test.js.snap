// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas adCategory Debe de mostrar correctamente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <p>
       
       
    </p>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
